### PR TITLE
Update libpopcnt to v2.6

### PR DESCRIPTION
**Pull request by Jvr2022**

### CODE REVIEW

File: oss/libpopcnt/libpopcnt.h [hunk 1]
```diff
@@ -95,7 +95,7 @@
   #define HAVE_AVX512
 #endif
 
-#if defined(X86_OR_X64) && !defined(_M_ARM64EC)
```
Comment (DHowett) [inline oss/libpopcnt/libpopcnt.h:98]: We should have documented this change, but we added ARM64EC because the Windows build system breaks otherwise.

**Review by DHowett — CHANGES_REQUESTED** [review]

File: oss/libpopcnt/libpopcnt.h (hunk 1 above)
Comment (Jvr2022) [reply oss/libpopcnt/libpopcnt.h:98]: so i need change this to ARM64EC?

File: oss/libpopcnt/libpopcnt.h [hunk 2]
```diff
@@ -95,7 +95,7 @@
...
+#if defined(X86_OR_X64)
```
Comment (zadjii-msft) [inline oss/libpopcnt/libpopcnt.h:98]:
```suggestion
#if defined(X86_OR_X64) && !defined(_M_ARM64EC)
```

File: oss/libpopcnt/libpopcnt.h (hunk 1 above)
Comment (zadjii-msft) [reply oss/libpopcnt/libpopcnt.h:98]: this needs to get changed back, yea

**Review by zadjii-msft — CHANGES_REQUESTED** [review]

**Comment by Jvr2022** [comment]
fixed

**Comment by zadjii-msft** [comment]
After some team discussion, we've come to the conclusion that we don't really think we can accept this PR right now. There's some pending internal work that we need to do to get all our OSS dependencies moved over to `vcpkg` (ala #15855). We can update OSS dependencies, but there's quite a lot of extra (internal) process at this time, that it doesn't make sense to continually ingest small version bumps that don't have any product-critical fixes in them. Considering the only change here is "Fix building on NetBSD" (https://github.com/kimwalisch/libpopcnt/releases/tag/v2.6), that's not really relevant to our product code 🤷

**Comment by Jvr2022** [comment]
Ok

**Pull request closed without merging**